### ISSUE TITLE
Consolidate artwork queries to using multiple routes

### ISSUE THREAD

**Issue by orta** [artsy/metaphysics#440]
There are multiple ways to query artworks on gravity, so we want to be able to handle them transparently behind the scenes depending on the fields in the query.

For example

```graphql
  {
    gene(id:"500-1000-ce") {
      id
      artworks(aggregations:[MEDIUM, PRICE_RANGE, TOTAL], sort:"prices", for_sale:false, page:1){
        hits {
          title
        }
      }
    }
  }
```

Would use the `filtered_artworks` query routes, but if you had something like:

```graphql
  {
    gene(id:"500-1000-ce") {
      id
      artworks(page:1){
        hits {
          title
        }
      }
    }
  }
```

It would handle the query on a different route, perhaps `api/v1/gene/artworks` for example. 

---

The tricky thing may be making sure responses conform to the same query shape afterwards. Additional [context in slack](https://artsy.slack.com/archives/metaphysics/p1476454697000166).

**Comment by alloy** [comment]
Not sure if this is what you mean by the tricky bit, but one issue I see is with the additional 'hits' field that hosts the artworks vs without the filter.

**Comment by alloy** [comment]
@orta @mzikherman Been thinking about the above mentioned issue a bit more, what about if we switch to the Relay [‘Connections’ specification](https://facebook.github.io/relay/graphql/connections.htm) for associations? It adds a bit more ceremony, but it does allow arbitrary metadata on _any_ association, such as e.g. [a count](http://stackoverflow.com/a/34192654). This would for the unfiltered and filtered artworks to both follow the same shapes.

**Comment by orta** [comment]
We're going to need this eventually, is it something we can do that's backwards compatible - e.g. if we release Wednesday, can the API stay the same?

**Comment by alloy** [comment]
Yeah then we can add it as the 'artworks' field end deprecate the 'filtered_artworks' field.
